### PR TITLE
[docs] Add documentation for writing cluster integration tests

### PR DESCRIPTION
**Pull request by nbroyles**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/integration/cluster_integration_tests.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+prom.Setup()
+defer prom.Close()
+
+// Run tests...
+
+
+```
+
+### Spinning up an individual component
+We've been mostly referring to a cluster with a coordinator, db node, and potentially an aggregator, but it's also possible to spin up each component individually. Each component has the same handful of constructors in the `inprocess` [package](https://github.com/m3db/m3/tree/master/src/integration/resources/inprocess). Here's an example of starting up a DB node.
```
Comment (wesleyk) [inline src/integration/cluster_integration_tests.md:201]: when would you do this out of curiosity?

**Review by wesleyk — APPROVED** [review]

File: src/integration/cluster_integration_tests.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+})
+```
+This configuration will spin up the following:
+
+* DB nodes with an RF of 3 and 1 instance for each RF (i.e. 3 separate invocations of M3DB in a single process)
+* 1 coordinator
+* Aggregators with an RF of 2 and 2 instances for each RF (i.e. 4 separate aggregator invocations in a single process)
+
+It's worth pointing out that you aren't required to spin up a full cluster each time. `inprocess.NewCluster` also allows you to just spin up a dbnode and coordinator, and later we demonstrate how to spin up just a single component.
+
```
Comment (fishie9) [inline src/integration/cluster_integration_tests.md:111]: do ou want to mention ClusterSpecification in this section?

File: src/integration/cluster_integration_tests.md (hunk 1 above)
Comment (nbroyles) [reply src/integration/cluster_integration_tests.md:201]: Most likely, you're debugging something in the write path of the DB node and you know you don't need a coordinator. So you could just spin up a DB node and hit it via the node APIs.

File: src/integration/cluster_integration_tests.md (hunk 2 above)
Comment (nbroyles) [reply src/integration/cluster_integration_tests.md:111]: I was going to leave that out for now as I think the use case is a bit niche (e.g. multi-cluster setups). If we get more questions about it we can definitely add.

**Pull request merged**